src/App.jsx

```diff
@@ -14,33 +16,95 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button type="button" className="button is-info is-light">
-        Sort alphabetically
-      </button>
+const SORT_BY_NAME = 'sort by name';
+const SORT_BY_LENGTH = 'sort by length';
 
-      <button type="button" className="button is-success is-light">
-        Sort by length
-      </button>
+const getSortedGoods = (goods, sortBy, isReverseAdded) => {
+  const sortedGoods = [...goods];
 
-      <button type="button" className="button is-warning is-light">
-        Reverse
-      </button>
+  if (sortBy) {
+    sortedGoods.sort((good1, good2) => {
+      switch (sortBy) {
+        case SORT_BY_NAME:
+          return good1.localeCompare(good2);
 
-      <button type="button" className="button is-danger is-light">
-        Reset
-      </button>
-    </div>
+        case SORT_BY_LENGTH:
+          return good1.length - good2.length;
+
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (isReverseAdded) {
+    return sortedGoods.reverse();
+  }
+
+  return sortedGoods;
+};
+
+export const App = () => {
+  const [sortBy, setSortBy] = useState(null);
+  const [isReverseAdded, setIsReverseAdded] = useState(false);
+
+  const visibleGoods = getSortedGoods(goodsFromServer, sortBy, isReverseAdded);
+  const isResetButtonVisible = isReverseAdded || sortBy;
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+  return (
+    <div className="section content">
+      <div className="buttons">
+        <button
+          type="button"
+          className={cn('button', 'is-info', {
+            'is-light': sortBy !== SORT_BY_NAME,
+          })}
+          onClick={() => setSortBy(SORT_BY_NAME)}
+        >
+          Sort alphabetically
+        </button>
+
+        <button
+          type="button"
+          className={cn('button', 'is-info', {
+            'is-light': sortBy !== SORT_BY_LENGTH,
+          })}
+          onClick={() => setSortBy(SORT_BY_LENGTH)}
+        >
+          Sort by length
+        </button>
+
+        <button
+          type="button"
+          className={cn('button', 'is-warning', {
+            'is-light': !isReverseAdded,
+          })}
+          onClick={() => setIsReverseAdded(isReverseAdded !== true)}
```

Comment:
I have no idea why did you do it `setIsReverseAdded(isReverseAdded !== true)`, but if you need to change the value depending on the previous state use functional setState, `setIsReverseAdded(prev => !prev)`

Reply:
I check here if isReverseAdded is false then setIsReverseAdded(true) and vice versa. 

It works with your example, but I don't understand it, I need to clarify on QnA.